### ISSUE TITLE
[std/testing] Function signature of assertObjectEquals() does not accept interfaces

### ISSUE THREAD

**Issue by Seally** [denoland/deno_std#741]
The type `Record<PropertyKey, unknown>` used in `assertObjectEquals()` does not accept interface types as an argument. This is due to an intentional quirk in how TypeScript operates with interfaces. With that said, I do not believe this assertion function should disallow inputs that uses types declared with the `interface` keyword.

```ts
import { assertObjectMatch } from "https://deno.land/std@0.87.0/testing/asserts.ts";

interface NumberToken {
    type: "number";
    value: number;
}

const token: NumberToken = {
    type: "number",
    value: 15
};

assertObjectMatch(token, { type: "number" }); // doesn't work

type TNumberToken = {
    type: "number";
    value: number;
};

const typeToken: TNumberToken = {
    type: "number",
    value: 15
};

assertObjectMatch(typeToken, { type: "number" }); // works
```

I'm not completely sure what to suggest changing this to, but the most minimal change is to use `Record<PropertyKey, any>` instead, which seems to work for this case and confirmed to not be a bug from the TypeScript team.

Here are some test inputs for the `Record<PropertyKey, any>` signature:

```ts
// Continued from previous block
function foo(input: Record<PropertyKey, any>) {}

foo(token); // pass
foo(typeToken); // pass
foo(foo); // pass
foo(""); // fail
foo(15); // fail
```

Relevant issues:
* https://github.com/microsoft/TypeScript/issues/42825
* https://github.com/microsoft/TypeScript/issues/41518

**Comment by cola119** [comment]
I'm facing this issue too. May I make the p-r to change the props type `Record<PropertyKey, unknown>` to `Record<PropertyKey, any>`? I think this change is trivial and there don't seem to be any unexpected problems.

**Comment by getspooky** [comment]
I fixed https://github.com/denoland/deno_std/pull/763

**Comment by wperron** [comment]
Fixed in #763
